### PR TITLE
Hide restricted ui elements, remove <br> from error message

### PR DESCRIPTION
**Pull request by jasnovak**
Only show explore and edit buttons for a slice if user has explore or edit access. Don't show save buttons on dashboard if user doesn't have save access.

Also, remove a "\<br\>" in the error message of the SaveModal. It displays on screen since the message is not html.

### CODE REVIEW

File: superset/assets/src/dashboard/components/SliceHeader.jsx [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
 import moment from 'moment';
+import { connect } from 'react-redux';
 
 import { t } from '../../locales';
 import EditableTitle from '../../components/EditableTitle';
 import TooltipWrapper from '../../components/TooltipWrapper';
 
 const propTypes = {
   slice: PropTypes.object.isRequired,
+  superset_can_explore: PropTypes.bool,
```
Comment (mistercrunch) [inline superset/assets/src/dashboard/components/SliceHeader.jsx]: Let's use camel case for props. The guideline is it's ok to reference objects keys coming out of python in underscore notation, but for everything declared on the js side (const, let, props) we use camel case.

File: superset/assets/src/dashboard/components/SliceHeader.jsx [hunk 2]
```diff
@@ -1,13 +1,16 @@
...
+  slice_can_edit: PropTypes.bool,
```
Comment (mistercrunch) [inline superset/assets/src/dashboard/components/SliceHeader.jsx]: here too

File: superset/assets/src/dashboard/components/SliceHeader.jsx (hunk 1 above)
Comment (jasnovak) [reply superset/assets/src/dashboard/components/SliceHeader.jsx]: Sounds good! Will do. Yes, there are other places where snake case is used in the js besides the examples I gave -- could use a pass to fix them all up for sure.

**Comment by jasnovak** [comment]
@mistercrunch Yes -- and there may be other places where this type of thing could be applied, i.e. not showing controls for which a user doesn't have access. I took care of these cases as they were important for us with our read-only, dashboard level access controls.

**Comment by williaster** [comment]
Yep I think we're on top of these in the new dashboard.

**Comment by jasnovak** [comment]
Yes, this is intended behavior. We have a read-only role that we don't want to give the ability to create dashboards. So we don't give them save access, which for this case should also mean no save-as.

**Comment by graceguo-supercat** [comment]
@jasnovak Thanks for clarify. Do you have some roles that can view and save-as other owner's dashboards, but cannot edit/overwrite changes on the original dashboard?

**Comment by jasnovak** [comment]
@graceguo-supercat No, we don't have that case. But I think to support both cases, you may need another permission, say can_save_as_dash

**Comment by graceguo-supercat** [comment]
@jasnovak In Superset build-in [roles](https://superset.apache.org/security.html#provided-roles): both Alpha user and Gamma user can save dashboard, and can save other owner's dashboard as their copy. [There are tests to assure this feature](https://github.com/apache/incubator-superset/blob/7f1d7543d05cedf5b5725b6da81943bd551ccc87/tests/security_tests.py#L48). We have a lot of users want to copy other's dashboard, but not change the original one. But with this PR, all Alpha and Gamma users can't save copy anymore.

I want to make some changes that still allow Alpha+Gamma users `save dashboard as copy`.  For the users in your system that don't have save access, are they any of build-in roles, or any role that removes can_save_dash? thanks!

**Comment by jasnovak** [comment]
Yes, we remove can_save_dash from our read-only users. So, if you want to make can_save_dash allow people to copy other's dashboards, that would work fine. Thanks for asking!

**Pull request merged**